rpmlb/builder/koji.py

```diff
@@ -130,33 +130,52 @@ def prepare_extra_steps(self, source, package_dict):
     def build(self, package_dict, **kwargs):
         """Build a package using Koji instance"""
 
+        base_name = package_dict['name']
+        full_name = self._full_package_name(base_name, self.collection)
         srpm_path = self._make_srpm(
-            name=package_dict['name'],
+            name=base_name,
             collection=self.collection,
             epel=self.epel,
         )
 
-        self._add_package(name='{collection}-{name}'.format(
-            collection=self.collection,
-            name=package_dict['name'],
-        ))
+        self._add_package(full_name)
         self._submit_build(srpm_path)
         self._wait_for_repo()
 
     @staticmethod
-    def _make_srpm(name: str, collection: str, epel: int) -> Path:
+    def _full_package_name(base_name: str, collection: str) -> str:
+        """Determine what the full name of the package should be
+        after applying the collection prefix.
+
+        Keyword arguments:
+            base_name: Name of the package, without the collection part.
+            collection: Name/identification of the package's collection.
+
+        Returns:
+            Canonical full name.
+        """
+
+        # The metapackage, or the base_name already has the prefix applied.
+        if base_name.startswith(collection):
```

Comment:
Should it not start with `colection+'-'`?

Reply:
No, because the name of the metapackage is exactly `collection`. `base_name` is not NVR, just the "N" part.

Reply:
So in fact it should be: `if base_name.startswith(collection + '-') or base_name == collection:`?

Reply:
Basically, yes. The important part is the `base_name == collection`. The `startswith` check is best-effort to avoid duplicate attachment of the prefix if the name already starts with it (and so in fact it is `full_name`).

If we are afraid of false positives here, I would drop the `startswith` check entirely.

Reply:
I'm just being paranoid. However I was bit by almost the same thing in the past. See https://github.com/pytest-dev/pytest/issues/2939#issuecomment-346207679

Reply:
I understand. The trouble here is that I'm not sure if the `'-'` at the end would help anyhow 😉.

Since the problem I'm trying to solve here manifests (so far) only on metapackages, I will drop the `startswith` check, which should make the condition unambiguous. It can be always added back if we run into a case that needs it.

Reply:
Do it. Thank you.